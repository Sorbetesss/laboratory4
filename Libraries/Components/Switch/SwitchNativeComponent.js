/**
<<<<<<< HEAD
 * Copyright (c) 2013-present, Facebook, Inc.
=======
 * Copyright (c) Facebook, Inc. and its affiliates.
>>>>>>> v0.58.6
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 * @format
 */

'use strict';

const Platform = require('Platform');
const ReactNative = require('ReactNative');

const requireNativeComponent = require('requireNativeComponent');

import type {SwitchChangeEvent} from 'CoreEventTypes';
import type {ViewProps} from 'ViewPropTypes';
<<<<<<< HEAD
import type {SemanticOrDynamicColorType} from 'normalizeColor' // TODO(macOS ISS#2323203)
=======
>>>>>>> v0.58.6

// @see ReactSwitchManager.java
export type NativeAndroidProps = $ReadOnly<{|
  ...ViewProps,
  enabled?: ?boolean,
  on?: ?boolean,
  onChange?: ?(event: SwitchChangeEvent) => mixed,
<<<<<<< HEAD
  thumbTintColor?: ?(string | SemanticOrDynamicColorType),
  trackTintColor?: ?(string | SemanticOrDynamicColorType),
=======
  thumbTintColor?: ?string,
  trackTintColor?: ?string,
>>>>>>> v0.58.6
|}>;

// @see RCTSwitchManager.m
export type NativeIOSProps = $ReadOnly<{|
  ...ViewProps,
  disabled?: ?boolean,
  onChange?: ?(event: SwitchChangeEvent) => mixed,
<<<<<<< HEAD
  onTintColor?: ?(string | SemanticOrDynamicColorType),
  thumbTintColor?: ?(string | SemanticOrDynamicColorType),
  tintColor?: ?(string | SemanticOrDynamicColorType),
=======
  onTintColor?: ?string,
  thumbTintColor?: ?string,
  tintColor?: ?string,
>>>>>>> v0.58.6
  value?: ?boolean,
|}>;

type SwitchNativeComponentType = Class<
  ReactNative.NativeComponent<
    $ReadOnly<{|
      ...NativeAndroidProps,
      ...NativeIOSProps,
    |}>,
  >,
>;

const SwitchNativeComponent: SwitchNativeComponentType =
  Platform.OS === 'android'
    ? (requireNativeComponent('AndroidSwitch'): any)
    : (requireNativeComponent('RCTSwitch'): any);

module.exports = SwitchNativeComponent;
