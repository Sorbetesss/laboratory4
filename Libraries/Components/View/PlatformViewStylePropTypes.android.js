/**
 * Copyright (c) 2015-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @providesModule PlatformViewStylePropTypes
 * @flow
 */

module.export = {
  /**
   * (Android-only) Sets the elevation of a view, using Android's underlying
   * [elevation API](https://developer.android.com/training/material/shadows-clipping.html#Elevation).
   * This adds a drop shadow to the item and affects z-order for overlapping views.
   * Only supported on Android 5.0+, has no effect on earlier versions.
   * @platform android
   */
  elevation: ReactPropTypes.number,
};
