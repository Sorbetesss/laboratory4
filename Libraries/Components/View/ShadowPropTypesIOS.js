/**
 * Copyright (c) 2015-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * @providesModule ShadowPropTypesIOS
 * @flow
 */
'use strict';

var ColorPropType = require('ColorPropType');
var ReactPropTypes = require('ReactPropTypes');

var ShadowPropTypesIOS = {
  /**
   * Sets the drop shadow color
   * @platform ios
   */
  shadowColor: ColorPropType,
  /**
   * Sets the drop shadow offset
   * height moves the shadow up (negative) and down (positive)
   * width moves the shadow left (negative) and right (positive)
   * @platform ios
   */
  shadowOffset: ReactPropTypes.shape(
    {width: ReactPropTypes.number, height: ReactPropTypes.number}
  ),
  /**
   * Sets the drop shadow opacity (multiplied by the color's alpha component)
   * 0 is transparent and 1 is opaque
   * @platform ios
   */
  shadowOpacity: ReactPropTypes.number,
  /**
   * Sets the drop shadow blur radius
   * Larger numbers have larger, but lighter shadow
   * 0 has a solid edge
   * @platform ios
   */
  shadowRadius: ReactPropTypes.number,
};

module.exports = ShadowPropTypesIOS;
