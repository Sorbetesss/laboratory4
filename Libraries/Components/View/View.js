/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @flow strict-local
 */

import type {ViewProps} from './ViewPropTypes';

import ViewNativeComponent from './ViewNativeComponent';
import TextAncestor from '../../Text/TextAncestor';
import * as React from 'react';

export type Props = ViewProps;

/**
 * The most fundamental component for building a UI, View is a container that
 * supports layout with flexbox, style, some touch handling, and accessibility
 * controls.
 *
 * @see https://reactnative.dev/docs/view
 */
const View: React.AbstractComponent<
  ViewProps,
  React.ElementRef<typeof ViewNativeComponent>,
> = React.forwardRef(
  (
    {tabIndex, focusable, role, accessibilityRole, ...otherProps}: ViewProps,
    forwardedRef,
  ) => {
    // Map role values to AccessibilityRole values
    const roleToAccessibilityRoleMapping = {
      slider: 'adjustable',
      img: 'image',
      presentation: 'none',
      summary: 'region',
    };

    const _accessibilityRole =
      roleToAccessibilityRoleMapping[role] ?? accessibilityRole;

    return (
      <TextAncestor.Provider value={false}>
        <ViewNativeComponent
          focusable={tabIndex !== undefined ? !tabIndex : focusable}
          accessibilityRole={_accessibilityRole}
          {...otherProps}
          ref={forwardedRef}
        />
      </TextAncestor.Provider>
    );
  },
);

View.displayName = 'View';

module.exports = View;
