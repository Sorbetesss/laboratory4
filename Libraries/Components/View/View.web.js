/**
 * @providesModule View
 */
'use strict';

var React = require('React');
var StyleSheet = require('StyleSheet');
var StyleSheetPropType = require('StyleSheetPropType');
var ViewStylePropTypes = require('ViewStylePropTypes');
var webifyStyle = require('webifyStyle');

var stylePropType = StyleSheetPropType(ViewStylePropTypes);

var pixelKeys = {
    top: true,
    left: true,
    right: true,
    bottom: true,
};

var View = React.createClass({

    propTypes: {
        style: stylePropType,
    },

    setNativeProps: function(props) {
        if (!this.refs.div) {
            return;
        }
        if (props.opacity) {
            this.refs.div.getDOMNode().style.opacity = props.opacity;
        }
        if (props.style) {
            var style = webifyStyle(props.style);
            for (var key in style) {
                var value = style[key];
                if (pixelKeys[key]) {
                    value = value + 'px';
                }
                this.refs.div.getDOMNode().style[key] = value;
            }
        }
    },

    measure: function(callback) {
        var m = this.refs.div.getDOMNode().getBoundingClientRect();
        if (callback) {
            callback(m.left, m.top, m.width, m.height, m.left, m.top);
        }
        return m;
    },

    render: function() {
        var style = webifyStyle(this.props.style);

        // I don't like this at all but
        // componentDidUpdate is not being fired reliably
        // Not sure what's going on...
        setTimeout(this._onLayout, 50);

        return (
            <div
                {...this.props}
                ref="div"
                style={style}
                onTouchStart={this._onTouchStart}
                onTouchMove={this._onTouchMove}
                onTouchEnd={this._onTouchEnd}
                onTouchCancel={this._onTouchCancel}
                onClick={this.props.onClick || this.props.onPress}
            />
        );
    },

    _onLayout: function() {
        if (!this.props.onLayout || !this.refs.div) {
            return;
        }
        var measure = this.measure();
        this.props.onLayout({
            nativeEvent: {
                layout: {
                    width: measure.width,
                    height: measure.height,
                },
            },
        });
    },

    _onTouchStart: function(e) {
        var domID = this.refs.div.getDOMNode().dataset.reactid;
        if (this.props.onResponderGrant) {
            this.props.onResponderGrant(e, domID);
        }
    },

    _onTouchMove: function(e) {
        if (this.props.onResponderMove) {
            this.props.onResponderMove(e);
        }
    },

    _onTouchEnd: function(e) {
        if (this.props.onResponderRelease) {
            this.props.onResponderRelease(e);
        }
    },

    _onTouchCancel: function(e) {
        if (this.props.onResponderTerminate) {
            this.props.onResponderTerminate(e);
        }
    },

});

module.exports = View;
