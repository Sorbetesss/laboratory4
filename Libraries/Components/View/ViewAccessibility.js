/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @flow strict
 */

'use strict';

import type {SyntheticEvent} from '../../Types/CoreEventTypes';

// This must be kept in sync with the AccessibilityRolesMask in RCTViewManager.m
export type AccessibilityRole =
  | 'none'
  | 'button'
  | 'link'
  | 'search'
  | 'image'
  | 'keyboardkey'
  | 'text'
  | 'adjustable'
  | 'imagebutton'
  | 'header'
  | 'summary'
  | 'alert'
  | 'checkbox'
  | 'combobox'
  | 'menu'
  | 'menubar'
  | 'menuitem'
  | 'progressbar'
  | 'radio'
  | 'radiogroup'
  | 'scrollbar'
  | 'spinbutton'
  | 'switch'
  | 'tab'
  | 'tablist'
  | 'timer'
  | 'toolbar';

// This must be kept in sync with the AccessibilityStatesMask in RCTViewManager.m
export type AccessibilityStates = $ReadOnlyArray<
  | 'disabled'
  | 'selected'
  | 'checked'
  | 'unchecked'
  | 'busy'
  | 'expanded'
  | 'collapsed'
  | 'hasPopup',
>;

// the info associated with an accessibility action
export type AccessibilityActionInfo = $ReadOnly<{
  name: string,
  label?: string,
}>;

// The info included in the event sent to onAccessibilityAction
export type AccessibilityActionEvent = SyntheticEvent<
  $ReadOnly<{
    actionName: string,
  }>,
>;

export type AccessibilityState = {
  disabled?: boolean,
  selected?: boolean,
  checked?: ?boolean | 'mixed',
  busy?: boolean,
  expanded?: boolean,
};

export type AccessibilityValueDescription = {
  /**
   * The minimum value of this component's range. (should be an integer)
   */
  minimum?: number,

  /**
   * The current value of this component's range. (should be an integer)
   */
  current?: number,

  /**
   * The maximum value of this component's range. (should be an integer)
   */
  maximum?: number,

  /**
   * A textual description of this component's value. (will override minimum, current, and maximum if set)
   */
  text?: string,
};
