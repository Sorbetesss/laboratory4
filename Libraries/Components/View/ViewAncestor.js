/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict
 * @format
 */

'use strict';

const React = require('react');

/**
 * Used to retrieve parent view accessibilityLiveRegion
 */
const ViewAncestorContext = (React.createContext(
  false,
): React$Context<$FlowFixMe>);
if (__DEV__) {
  ViewAncestorContext.displayName = 'ViewAncestorContext';
}
module.exports = ViewAncestorContext;
