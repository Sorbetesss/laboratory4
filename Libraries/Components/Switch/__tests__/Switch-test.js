/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @emails oncall+react_native
 * @flow strict-local
 */

import * as React from 'react';

import Switch from '../Switch';
import View from '../../View/View';
import {expectRendersMatchingSnapshot} from '../../../Utilities/ReactNativeTestTools';

describe('<Switch />', () => {
  it('should render as expected', () => {
    expectRendersMatchingSnapshot(
      'Switch',
      () => (
        <Switch>
          <View />
        </Switch>
      ),
      () => {
        jest.dontMock('../Switch');
      },
    );
  });
});

describe('<Switch disabled={true} />', () => {
  it('should be disabled when disabled is true', () => {
    expectRendersMatchingSnapshot(
      'Switch',
      () => (
        <Switch disabled={true}>
          <View />
        </Switch>
      ),
      () => {
        jest.dontMock('../Switch');
      },
    );
  });
});

describe('<Switch disabled={true} accessibilityState={{}} />', () => {
  it('should be disabled when disabled is true and accessibilityState is empty', () => {
    expectRendersMatchingSnapshot(
      'Switch',
      () => (
        <Switch disabled={true} accessibilityState={{}}>
          <View />
        </Switch>
      ),
      () => {
        jest.dontMock('../Switch');
      },
    );
  });
});

describe('<Switch disabled={true} accessibilityState={{checked: true}} />', () => {
  it('should keep accessibilityState when disabled is true', () => {
    expectRendersMatchingSnapshot(
      'Switch',
      () => (
        <Switch disabled={true} accessibilityState={{checked: true}}>
          <View />
        </Switch>
      ),
      () => {
        jest.dontMock('../Switch');
      },
    );
  });
});

describe('<Switch disabled={true} accessibilityState={{disabled: false}} />', () => {
  it('should overwrite accessibilityState with value of disabled prop', () => {
    expectRendersMatchingSnapshot(
      'Switch',
      () => (
        <Switch disabled={true} accessibilityState={{disabled: false}}>
          <View />
        </Switch>
      ),
      () => {
        jest.dontMock('../Switch');
      },
    );
  });
});
