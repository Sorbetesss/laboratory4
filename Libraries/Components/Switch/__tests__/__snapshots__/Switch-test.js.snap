// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> should render as expected: should deep render when mocked (please verify output manually) 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  value={false}
>
  <View />
</RCTSwitch>
`;

exports[`<Switch /> should render as expected: should deep render when not mocked (please verify output manually) 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  value={false}
>
  <View />
</RCTSwitch>
`;

exports[`<Switch /> should render as expected: should shallow render as <Switch /> when mocked 1`] = `
<Switch>
  <View />
</Switch>
`;

exports[`<Switch /> should render as expected: should shallow render as <Switch /> when not mocked 1`] = `
<Switch>
  <View />
</Switch>
`;

exports[`<Switch disabled={true} /> should be disabled when disabled is true: should deep render when mocked (please verify output manually) 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  disabled={true}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  value={false}
>
  <View />
</RCTSwitch>
`;

exports[`<Switch disabled={true} /> should be disabled when disabled is true: should deep render when not mocked (please verify output manually) 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  disabled={true}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  value={false}
>
  <View />
</RCTSwitch>
`;

exports[`<Switch disabled={true} /> should be disabled when disabled is true: should shallow render as <Switch /> when mocked 1`] = `
<Switch
  disabled={true}
>
  <View />
</Switch>
`;

exports[`<Switch disabled={true} /> should be disabled when disabled is true: should shallow render as <Switch /> when not mocked 1`] = `
<Switch
  disabled={true}
>
  <View />
</Switch>
`;

exports[`<Switch disabled={true} accessibilityState={{}} /> should be disabled when disabled is true and accessibilityState is empty: should deep render when mocked (please verify output manually) 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  disabled={true}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  value={false}
>
  <View />
</RCTSwitch>
`;

exports[`<Switch disabled={true} accessibilityState={{}} /> should be disabled when disabled is true and accessibilityState is empty: should deep render when not mocked (please verify output manually) 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  disabled={true}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  value={false}
>
  <View />
</RCTSwitch>
`;

exports[`<Switch disabled={true} accessibilityState={{}} /> should be disabled when disabled is true and accessibilityState is empty: should shallow render as <Switch /> when mocked 1`] = `
<Switch
  accessibilityState={Object {}}
  disabled={true}
>
  <View />
</Switch>
`;

exports[`<Switch disabled={true} accessibilityState={{}} /> should be disabled when disabled is true and accessibilityState is empty: should shallow render as <Switch /> when not mocked 1`] = `
<Switch
  accessibilityState={Object {}}
  disabled={true}
>
  <View />
</Switch>
`;

exports[`<Switch disabled={true} accessibilityState={{checked: true}} /> should keep accessibilityState when disabled is true: should deep render when mocked (please verify output manually) 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  accessibilityState={
    Object {
      "checked": true,
      "disabled": true,
    }
  }
  disabled={true}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  value={false}
>
  <View />
</RCTSwitch>
`;

exports[`<Switch disabled={true} accessibilityState={{checked: true}} /> should keep accessibilityState when disabled is true: should deep render when not mocked (please verify output manually) 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  accessibilityState={
    Object {
      "checked": true,
      "disabled": true,
    }
  }
  disabled={true}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  value={false}
>
  <View />
</RCTSwitch>
`;

exports[`<Switch disabled={true} accessibilityState={{checked: true}} /> should keep accessibilityState when disabled is true: should shallow render as <Switch /> when mocked 1`] = `
<Switch
  accessibilityState={
    Object {
      "checked": true,
    }
  }
  disabled={true}
>
  <View />
</Switch>
`;

exports[`<Switch disabled={true} accessibilityState={{checked: true}} /> should keep accessibilityState when disabled is true: should shallow render as <Switch /> when not mocked 1`] = `
<Switch
  accessibilityState={
    Object {
      "checked": true,
    }
  }
  disabled={true}
>
  <View />
</Switch>
`;

exports[`<Switch disabled={true} accessibilityState={{disabled: false}} /> should overwrite accessibilityState with value of disabled prop: should deep render when mocked (please verify output manually) 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  disabled={true}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  value={false}
>
  <View />
</RCTSwitch>
`;

exports[`<Switch disabled={true} accessibilityState={{disabled: false}} /> should overwrite accessibilityState with value of disabled prop: should deep render when not mocked (please verify output manually) 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  disabled={true}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 31,
      "width": 51,
    }
  }
  value={false}
>
  <View />
</RCTSwitch>
`;

exports[`<Switch disabled={true} accessibilityState={{disabled: false}} /> should overwrite accessibilityState with value of disabled prop: should shallow render as <Switch /> when mocked 1`] = `
<Switch
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  disabled={true}
>
  <View />
</Switch>
`;

exports[`<Switch disabled={true} accessibilityState={{disabled: false}} /> should overwrite accessibilityState with value of disabled prop: should shallow render as <Switch /> when not mocked 1`] = `
<Switch
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  disabled={true}
>
  <View />
</Switch>
`;
